src/treq/_utils.py

```diff
@@ -45,3 +48,50 @@ def default_pool(reactor, pool, persistent):
         set_global_pool(HTTPConnectionPool(reactor, persistent=True))
 
     return get_global_pool()
+
+
+def warn(*args, **kwargs):
```

Comment:
The `try`/`finally` was very intentional.  This is doing some goofy stuff, which maybe you tidied up a bit, but what we really don't want to have happen is have our attempt to warn someone of a future exception...  turn into another exception without the warning.  Speaking of which, I probably should have had an `except Exception: pass` to avoid any exception caused by this inspection ever.